src/frontend/mame/ui/menu.cpp

```diff
@@ -321,15 +321,25 @@ void menu::item_append(menu_item item)
 void menu::item_append(menu_item_type type)
 {
 	if (type == menu_item_type::SEPARATOR)
-		item_append(MENU_SEPARATOR_ITEM, nullptr, 0, nullptr, menu_item_type::SEPARATOR);
+		item_append(MENU_SEPARATOR_ITEM, "", 0, nullptr, menu_item_type::SEPARATOR);
 }
 
 //-------------------------------------------------
 //  item_append - append a new item to the
 //  end of the menu
 //-------------------------------------------------
 
-void menu::item_append(const char *text, const char *subtext, UINT32 flags, void *ref, menu_item_type type)
+void menu::item_append(const std::string &text, const std::string &subtext, UINT32 flags, void *ref, menu_item_type type)
+{
+	item_append((std::string&&)text, (std::string&&)subtext, flags, ref, type);
+}
+
+//-------------------------------------------------
+//  item_append - append a new item to the
+//  end of the menu
+//-------------------------------------------------
+
+void menu::item_append(std::string &&text, std::string &&subtext, UINT32 flags, void *ref, menu_item_type type)
```

Comment:
You need to actually std::move the rvalue references when assigning them, otherwise it just calls the lvalue version of the assignment operator anyway.